### PR TITLE
feat(wallet): always enable OKX and gate it on version >= 4.5.0

### PR DESCRIPTION
**Pull request by jrwbabylonlab**
- Remove OKX from the vault's opt-in wallet list so it's always shown
  (only utila stays behind NEXT_PUBLIC_TBV_EXTRA_BTC_WALLETS).
- Add a connect-time minimum-version gate (>= 4.5.0) in the wallet-connector
  OKX provider, mirroring the audited UniSat/OneKey pattern: reads root
  okxwallet.getVersion() via withTimeout, compares with checkMinVersion
  (strict semver — not the string compare removed in #740), and fails closed
  on a missing / unreadable / non-canonical version.
- Add okxVersion.test.ts (20 cases, parity with unisat/onekey).

### CODE REVIEW

File: services/vault/src/context/wallet/VaultWalletConnectionProvider.tsx [hunk 1]
```diff
@@ -21,22 +21,18 @@ import featureFlags from "@/config/featureFlags";
...
+// keeps non-conforming adapters out of the connect UI.
 const ALWAYS_DISABLED_WALLETS: string[] = [
   APPKIT_BTC_CONNECTOR_ID,
   "injectable",
   "ledger_btc",
   "ledger_btc_v2",
 ];
 
-const OPT_IN_WALLETS = ["okx", "utila"];
+const OPT_IN_WALLETS = ["utila"];
```
[the remark this answers is not in the document]
Comment (jrwbabylonlab) [reply services/vault/src/context/wallet/VaultWalletConnectionProvider.tsx:35]: Verified against OKX 4.5.0 (the version this gate requires): `okxwallet.bitcoin.connect()` returns `{ address, publicKey, compressedPublicKey }` — OKX's own `getCompressedPublicKey` reads `compressedPublicKey` off the standard `bitcoin.connect()` (only Fractal differs). Reading it here is correct; connect isn't broken. This parsing is also pre-existing — this PR only adds the version check.

File: packages/babylon-wallet-connector/src/core/wallets/btc/okx/provider.ts [hunk 2]
```diff
@@ -14,8 +16,13 @@ const PROVIDER_NAMES = {
 
 export const WALLET_PROVIDER_NAME = "OKX";
 
+// Bound the version read so a locked/asleep extension fails recoverably, not hangs.
+const OKX_RPC_TIMEOUT_MS = 10_000;
+
 export class OKXProvider implements IBTCProvider {
   private provider: any;
+  // Root `okxwallet` (getVersion lives here); `this.provider` is the per-chain provider.
+  private wallet: any;
```
[the remark this answers is not in the document]
Comment (jrwbabylonlab) [reply packages/babylon-wallet-connector/src/core/wallets/btc/okx/provider.ts:25]: Keeping `any` for consistency with the adjacent `private provider: any` and the other BTC providers (unisat/onekey), which all treat the injected wallet as untyped. Typing only `getVersion` here would be inconsistent without buying real safety.

**Review by gbarkhatov — DISMISSED** [review]

**Review by jonybur — COMMENTED** [review]
The version-comparison logic is solid - `checkMinVersion` handles all the semver edge cases (4.10.0 > 4.5.0, two-part, v-prefix, prerelease, leading zeros, undefined, non-string), it fails closed throughout, the 20 tests match behavior, and there's no dead code (OPT_IN_WALLETS correctly reduced to utila). Two assumptions to confirm before merge - inline.

File: packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { checkMinVersion, type MinVersionCheck } from "@/core/utils/checkMinVersion";
+
+// Floor for `deriveContextHash`, which vault deposits require; older OKX can't
+// deposit. Team-specified pending the exact OKX release ref. Compared via
+// checkMinVersion (strict semver), never a string `<` (the #740 Immunefi bug).
+export const MIN_OKX_VERSION = "4.5.0";
```
Comment (jonybur) [inline packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts:7]: **Should-fix - `MIN_OKX_VERSION` isn't tied to a real OKX release.** The comment admits it's "team-specified pending the exact OKX release ref." The UniSat and OneKey floors both pin the exact upstream commit/tag where `deriveContextHash` landed. If the true floor differs, this either admits older broken builds or rejects working ones. Worth blocking on confirming the actual version before merge.

File: packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+/** Compares a raw OKX `getVersion()` value against {@link MIN_OKX_VERSION}. */
+export function checkOKXVersion(raw: unknown): MinVersionCheck {
```
Comment (jonybur) [inline packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts:10]: **Should-fix - confirm OKX's `getVersion()` returns the extension app version.** `provider.ts` reads `this.wallet.getVersion()` and feeds it here. The OneKey gate exists precisely because `getVersion()` there returns a hardcoded package value (`1.4.10`) unrelated to the app version, which is why OneKey reads `$walletInfo.version` instead. If OKX's `getVersion()` is similarly hardcoded or returns the inpage-provider package version, this gate is either always-fail-closed (locks out all OKX users) or gates on the wrong number. Please confirm against a real OKX extension, since this gates `deriveContextHash` support.

File: packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts (hunk 3 above)
Comment (jrwbabylonlab) [reply packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts:7]: Confirmed against OKX 4.5.0: `window.okxwallet.getVersion()` returns `'4.5.0'` (matches the extension version) and `typeof window.okxwallet.bitcoin.deriveContextHash` is `'function'`. Will update the comment to cite this as the verified floor.

File: packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts (hunk 4 above)
Comment (jrwbabylonlab) [reply packages/babylon-wallet-connector/src/core/wallets/btc/okx/version.ts:10]: Verified: `okxwallet.getVersion()` returns the real extension version (`'4.5.0'`), not a hardcoded inpage-provider value like OneKey. Unlike OneKey's `getVersion()` which returns `'1.4.10'` regardless of app version, OKX's tracks the actual installed version.

**Review by gbarkhatov — APPROVED** [review]
